opendatahub-io/odh-dashboard
# [RHOAIENG-2980] Create run param types

Closes: [RHOAIENG-2980](https://issues.redhat.com/browse/RHOAIENG-2980)

## Description
Input parameters are no longer of just type "text", and reflect all possible types. Note that the design was adjusted based on conversation with our designers to have JSON fields be TextArea inputs. Also, I spoke with PF design about the width of number inputs, and those with +/- symbols were not meant to be at a greater width than that of the characters contained in the middle text box.

## How Has This Been Tested?
Cypress test with all parameter types edited and submitted with verified payload. Regression tested with new run creation, duplication which populates all input params with fetched values.

**Create run**
<img width="572" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/96431149/914d47e4-c4b0-41e1-b9d7-2538c759b20d">

**Run details**
<img width="572" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/96431149/33d54ebd-785e-47f3-bf34-832eb0fe6038">

## How to test
Create a new run or schedule with a pipeline that has input parameter definitions. Duplicate that run or schedule and verify parameter inputs are populated accordingly.

Self checklist (all need to be checked):

If you have UI changes: 

After the PR is posted & before it merges:
